src/engine/Server.cpp

```diff
@@ -757,13 +757,10 @@ boost::asio::awaitable<void> Server::processQuery(
 // _____________________________________________________________________________
 template <typename Function, typename T>
 Awaitable<T> Server::computeInNewThread(Function function) const {
-  auto acquireComputeRelease = [this, function = std::move(function)] {
-    LOG(DEBUG) << "Acquiring new thread for query processing\n";
-    queryProcessingSemaphore_.acquire();
-    auto cleanup = ad_utility::makeOnDestructionDontThrowDuringStackUnwinding(
-        [this]() { queryProcessingSemaphore_.release(); });
-    return function();
+  auto runOnExecutor = [](auto executor, Function func) -> net::awaitable<T> {
+    co_await net::post(executor, net::use_awaitable);
+    co_return std::invoke(func);
   };
-  co_return co_await ad_utility::asio_helpers::async_on_external_thread(
-      std::move(acquireComputeRelease), boost::asio::use_awaitable);
+  return ad_utility::sameExecutor(
+      runOnExecutor(threadPool_.get_executor(), std::move(function)));
```

Comment:
In the case of an exception in the inner query processing we still send Messages etc. so maybe we now need a guarantee that we are back on the same executor even in the case of error.